### ISSUE TITLE
Receiver becomes unresponsive to network

### ISSUE THREAD

**Issue by chetstone** [grzegorz914/homebridge-denon-tv#114]
Every couple of days the receiver stops communicating with Homekit (or with the Marantz AVR). And sometimes it even becomes unresponsive to IR Remote control.
I have to unplug the receiver , wait thirty seconds and plug it back in to get it working again. 
 
I can't say for sure what's causing it, but I think it's very weird that this component is continually sending messages to the receiver every five seconds or so. I can't think of a legitimate reason for this. Version 3.11.3

```
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>MPLAY</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>-39.5</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:11 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>SOURCE</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>  0</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:16 AM] [DenonTv] Device: 192.168.1.208 Marantz Main Zone, requesting Device state.
[9/22/2021, 12:51:16 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2 Zone 2, requesting Device state.
[9/22/2021, 12:51:16 AM] [DenonTv] Device: 192.168.1.208 Marantz, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>MPLAY</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>-39.5</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:16 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>SOURCE</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>  0</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:21 AM] [DenonTv] Device: 192.168.1.208 Marantz Main Zone, requesting Device state.
[9/22/2021, 12:51:21 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2 Zone 2, requesting Device state.
[9/22/2021, 12:51:21 AM] [DenonTv] Device: 192.168.1.208 Marantz, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>MPLAY</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>-39.5</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:21 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>SOURCE</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>  0</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:26 AM] [DenonTv] Device: 192.168.1.208 Marantz Main Zone, requesting Device state.
[9/22/2021, 12:51:26 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2 Zone 2, requesting Device state.
[9/22/2021, 12:51:26 AM] [DenonTv] Device: 192.168.1.208 Marantz, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>MPLAY</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>-39.5</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:26 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>SOURCE</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>  0</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:31 AM] [DenonTv] Device: 192.168.1.208 Marantz Main Zone, requesting Device state.
[9/22/2021, 12:51:31 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2 Zone 2, requesting Device state.
[9/22/2021, 12:51:31 AM] [DenonTv] Device: 192.168.1.208 Marantz, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>MPLAY</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>-39.5</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:31 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>SOURCE</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>  0</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:36 AM] [DenonTv] Device: 192.168.1.208 Marantz Main Zone, requesting Device state.
[9/22/2021, 12:51:36 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2 Zone 2, requesting Device state.
[9/22/2021, 12:51:36 AM] [DenonTv] Device: 192.168.1.208 Marantz, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>MPLAY</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>-39.5</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
[9/22/2021, 12:51:36 AM] [DenonTv] Device: 192.168.1.208 Marantz Zone 2, debug response: <?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<InputFuncSelect><value>SOURCE</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>  0</value></MasterVolume>
<Mute><value>off</value></Mute>
</item>
, result: { item: [Object] }
```

**Comment by grzegorz914** [comment]
All is OK here, if You want You can change refresh time of receiver state in plugin config. 
I have not observe any problems on my side, also in the log all is OK, this must be specific problem on your network or receiver.

**Comment by chetstone** [comment]
Thanks for the help. I think the problem was related to HDMI-CEC rather than this component or anything to do with IP networking. I had two input devices set up with HDMI Passthrough so when I turned one of them on it should switch the receiver input and turn on the TV. But often when I turned one on, the receiver would get really confused and switch inputs to the wrong one. So I just disabled HDMI-CEC on the older of the two devices and I haven't had any problem with the receiver hanging for the last week. Now if I want to use that device I have to switch the input and turn on the TV manually but it's not a big deal.
